### PR TITLE
feat: add types for NuxtConfigurationRender

### PR DESCRIPTION
**Pull request by danielroe**

### CODE REVIEW

File: packages/types/config/render.d.ts [hunk 1]
```diff
@@ -7,20 +7,65 @@
...
+  csp?: boolean | CspOptions
   dist?: ServeStaticOptions
-  etag?: EtagOptions | false
-  fallback?: any // https://github.com/nuxt/serve-placeholder types TBD
-  http2?: any // TBD
+  etag?: NuxtEtagOptions | false
+  fallback?: {
+    dist?: ServePlaceholderOptions
+    static?: ServePlaceholderOptions
+  };
```
Comment (kevinmarrec) [inline packages/types/config/render.d.ts]:
```suggestion
  }
```

File: packages/types/config/render.d.ts [hunk 2]
```diff
@@ -7,20 +7,65 @@
...
+  http2?: {
+    push?: boolean
+    shouldPush?: boolean
+    pushAssets?: (
+      req: IncomingMessage,
+      res: ServerResponse,
+      publicPath: string,
+      preloadFiles: PreloadFile[]
+    ) => string[]
+  };
```
Comment (kevinmarrec) [inline packages/types/config/render.d.ts]:
```suggestion
  }
```

File: packages/types/config/render.d.ts [hunk 3]
```diff
@@ -7,20 +7,65 @@
...
+  handlers?: Record<string, ServePlaceholderHandler | null | false>
+  mimes?: Record<ServePlaceholderHandler, string | false | undefined>
+  noCache?: boolean
+  placeholders?: Record<ServePlaceholderHandler, string | Buffer | false>
+  skipUnknown?: boolean
+  statusCode?: false | number
+}
+
+type CspPolicyName = 'child-src' | 'connect-src' | 'default-src' | 'font-src' | 'frame-src' | 'img-src' | 'manifest-src' | 'media-src' | 'object-src' | 'prefetch-src' | 'script-src' | 'script-src-elem' | 'script-src-attr' | 'style-src' | 'style-src-elem' | 'style-src-attr' | 'worker-src' | 'base-uri' | 'plugin-types' | 'sandbox' | 'form-action' | 'frame-ancestors' | 'navigate-to' | 'report-uri' | 'report-to' | 'block-all-mixed-content' | 'referrer' | 'require-sri-for' | 'trusted-types' | 'upgrade-insecure-requests'
+interface CspOptions {
```
Comment (kevinmarrec) [inline packages/types/config/render.d.ts]: Are `hashAlgorithm` & `reportOnly` mandatory options or are they missing `?` if they are not ?

Comment (danielroe) [reply packages/types/config/render.d.ts]: They are wrongly missing ?.

File: packages/types/config/render.d.ts [hunk 4]
```diff
@@ -7,20 +7,65 @@
  *                https://github.com/jshttp/etag#readme
  */
 
+import { IncomingMessage, ServerResponse } from 'http'
```
Comment (kevinmarrec) [inline packages/types/config/render.d.ts]: Probably relevant to move this line down under `etag` imports ones (alphabetical)

Comment (danielroe) [reply packages/types/config/render.d.ts]: I think because it's a node built-in, it should be at top.

Comment (kevinmarrec) [reply packages/types/config/render.d.ts]: Yeah tha'ts right

File: packages/types/config/render.d.ts [hunk 5]
```diff
@@ -7,20 +7,65 @@
...
+  noCache?: boolean
+  placeholders?: Record<ServePlaceholderHandler, string | Buffer | false>
+  skipUnknown?: boolean
+  statusCode?: false | number
+}
+
+type CspPolicyName = 'child-src' | 'connect-src' | 'default-src' | 'font-src' | 'frame-src' | 'img-src' | 'manifest-src' | 'media-src' | 'object-src' | 'prefetch-src' | 'script-src' | 'script-src-elem' | 'script-src-attr' | 'style-src' | 'style-src-elem' | 'style-src-attr' | 'worker-src' | 'base-uri' | 'plugin-types' | 'sandbox' | 'form-action' | 'frame-ancestors' | 'navigate-to' | 'report-uri' | 'report-to' | 'block-all-mixed-content' | 'referrer' | 'require-sri-for' | 'trusted-types' | 'upgrade-insecure-requests'
+interface CspOptions {
+  addMeta?: boolean
+  allowedSources?: {}
```
Comment (kevinmarrec) [inline packages/types/config/render.d.ts]: Why `{}` ?

Comment (danielroe) [reply packages/types/config/render.d.ts]: Sorry, that was incomplete. Have updated to `string[]` which I think is correct though note what appears to be an incorrect test [here](https://github.com/nuxt/nuxt.js/blob/257ae2279b142962de928902fedfda29cb94bb7d/packages/config/test/options.test.js#L101).

Comment (kevinmarrec) [reply packages/types/config/render.d.ts]: Hm alright, from what did you suppose it is `string[]` then ?

Comment (danielroe) [reply packages/types/config/render.d.ts]: [here](https://github.com/nuxt/nuxt.js/blob/4e4aa4d5d488a25bd2b78258c11b114808c9ea3d/packages/server/src/middleware/nuxt.js#L130), [here](https://github.com/nuxt/nuxt.js/blob/257ae2279b142962de928902fedfda29cb94bb7d/test/unit/basic.ssr.csp.test.js#L84) and [here](https://github.com/nuxt/nuxt.js/blob/4e4aa4d5d488a25bd2b78258c11b114808c9ea3d/packages/server/test/middleware/nuxt.test.js#L243)

Comment (kevinmarrec) [reply packages/types/config/render.d.ts]: Cool thanks !

**Pull request merged**